File: app/src/main/java/com/yao/zhihudaily/tool/OnItemClickListener.java
package com.yao.zhihudaily.tool;

import android.view.View;

/**
 *
 * @author Yao
 * @date 2016/7/28
 */
public abstract class OnItemClickListener implements View.OnClickListener, View.OnLongClickListener{

    public void onItemClick(int pos) {

    }

    public void onItemLongClick(int pos) {

    }

    @Override
    public void onClick(View view) {
        int pos = (int) view.getTag();
        onItemClick(pos);
    }

    @Override
    public boolean onLongClick(View view) {
        int pos = (int) view.getTag();
        onItemLongClick(pos);
        return false;
    }
}

File: app/src/main/java/com/yao/zhihudaily/tool/OnItemClickListener.kt
package com.yao.zhihudaily.tool

import android.view.View

/**
 *
 * @author Yao
 * @date 2016/7/28
 */
abstract class OnItemClickListener : View.OnClickListener, View.OnLongClickListener {

    open fun onItemClick(pos: Int) {

    }

    fun onItemLongClick(pos: Int) {

    }

    override fun onClick(view: View) {
        val pos = view.tag as Int
        onItemClick(pos)
    }

    override fun onLongClick(view: View): Boolean {
        val pos = view.tag as Int
        onItemLongClick(pos)
        return false
    }
}
